CONTRIBUTING.md

```diff
@@ -0,0 +1,132 @@
+2. Make your code changes in the new branch.
+3. Write clear and concise commit messages for each logical change.
+4. Make sure your code follows the project's coding conventions and style guidelines.
+5. Write unit tests to cover your code changes and ensure they pass.
+6. Update the relevant documentation, including README files and code comments, if necessary.
+7. Push your changes to your forked repository.
+8. Open a pull request against the main repository's `main` branch.
+9. Provide a clear and descriptive title for the pull request.
+10. Include a detailed description of the changes made and the problem they solve.
+11. Reference any related issues or feature requests in the pull request description.
+
+### Getting Started:
+
+- Fork this repo
+- Clone on your local machine
+
+```terminal
+git clone https://github.com/oss-slu/shelter_volunteers.git
+```
+
+- Create a new Branch
+
+```
+git checkout -b my-new-branch
+```
+- Add your changes
+```
+git add .
+```
+- Commit your changes.
+
+```
+git commit -m "Relevant message"
+```
+- Then push 
+```
+git push origin my-new-branch
+```
+
+
+- Create a new pull request from your forked repository
+
+<br>
+
+### Avoid Conflicts {Syncing your fork}
+
+An easy way to avoid conflicts is to add an 'upstream' for your git repo, as other PR's may be merged while you're working on your branch/fork.   
+
+```
+git remote add upstream https://github.com/oss-slu/shelter_volunteers
+```
+
+You can verify that the new remote has been added by typing
+```
+git remote -v
+```
+
+To pull any new changes from your parent repo simply run
+```
+git merge upstream/main
+```
+
+This will give you any eventual conflicts and allow you to easily solve them in your repo. It's a good idea to use it frequently in between your own commits to make sure that your repo is up to date with its parent.
+
+## Documenting Changes
+
+When making changes to the codebase, it's important to keep the documentation up to date. Here are some guidelines for documenting your changes:
+
+- Update the relevant README files, documentation, or user guides to reflect the changes made.
+- Add or update code comments to explain complex or non-obvious parts of the code.
+- Include any necessary configuration or setup instructions for new features or changes.
+
+## Writing Unit Tests
+
+We encourage contributors to write unit tests for their code changes to ensure the stability and reliability of the project. When writing unit tests:
+
+- Use the project's testing framework and conventions.
+- Write tests that cover the various scenarios and edge cases related to your code changes.
+- Ensure that the tests pass successfully before submitting a pull request.
+
+## Contacting the Team
+
+If you have any questions, concerns, or need further assistance, you can reach out to the project team through the following channel:
+
+[Slack channel](https://join.slack.com/t/oswslu/shared_invite/zt-24f0qhjbo-NkSfQ4LOg5wXxBdxP4vzfA)
```

Comment:
This is the generic link to join our Slack instance. You should be able to create a custom link that puts people directly in #project_shelter_volunteers, or you can link to this to join slack, and refer to joining the project channel.